api/v1/mongodbcommunity_types.go

```diff
@@ -119,6 +119,69 @@ type MongoDBCommunitySpec struct {
 	// Prometheus configurations.
 	// +optional
 	Prometheus *Prometheus `json:"prometheus,omitempty"`
+
+	// Additional options to be appended to the connection string. These options apply to the entire resource and to each user.
+	// +kubebuilder:validation:Type=object
+	// +optional
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +nullable
+	AdditionalConnectionStringConfig MapWrapper `json:"additionalConnectionStringConfig,omitempty"`
```

Comment:
Would it be simpler if we put a `map[string]interface{}` directly there?

Reply:
And what's the error then (out of curiosity)? I think it just requires a policy how to merge entries but I might be wrong.

Reply:
I did not try that actually. But according to this open issue: https://github.com/kubernetes-sigs/controller-tools/issues/636 and the comments that were already in the code, it should not work.